tfhe/src/core_crypto/fft_128_impl/crypto/bootstrap.rs

```diff
@@ -0,0 +1,384 @@
+        Fourier128LweBootstrapKey {
+            data_re0: self.data_re0.as_ref(),
+            data_re1: self.data_re1.as_ref(),
+            data_im0: self.data_im0.as_ref(),
+            data_im1: self.data_im1.as_ref(),
+            polynomial_size: self.polynomial_size,
+            input_lwe_dimension: self.input_lwe_dimension,
+            glwe_size: self.glwe_size,
+            decomposition_base_log: self.decomposition_base_log,
+            decomposition_level_count: self.decomposition_level_count,
+        }
+    }
+
+    pub fn as_mut_view(&mut self) -> Fourier128LweBootstrapKey<&mut [C::Element]>
+    where
+        C: AsMut<[C::Element]>,
+    {
+        Fourier128LweBootstrapKey {
+            data_re0: self.data_re0.as_mut(),
+            data_re1: self.data_re1.as_mut(),
+            data_im0: self.data_im0.as_mut(),
+            data_im1: self.data_im1.as_mut(),
+            polynomial_size: self.polynomial_size,
+            input_lwe_dimension: self.input_lwe_dimension,
+            glwe_size: self.glwe_size,
+            decomposition_base_log: self.decomposition_base_log,
+            decomposition_level_count: self.decomposition_level_count,
+        }
+    }
+}
+
+pub type FourierLweBootstrapKeyOwned = Fourier128LweBootstrapKey<ABox<[f64]>>;
+
+impl Fourier128LweBootstrapKey<ABox<[f64]>> {
+    pub fn new(
+        input_lwe_dimension: LweDimension,
+        polynomial_size: PolynomialSize,
+        glwe_size: GlweSize,
+        decomposition_base_log: DecompositionBaseLog,
+        decomposition_level_count: DecompositionLevelCount,
+    ) -> Fourier128LweBootstrapKey<ABox<[f64]>> {
+        let container_len = polynomial_size.to_fourier_polynomial_size().0
+            * input_lwe_dimension.0
+            * decomposition_level_count.0
+            * glwe_size.0
+            * glwe_size.0;
+
+        let boxed_re0 = avec![0.0f64; container_len].into_boxed_slice();
+        let boxed_re1 = avec![0.0f64; container_len].into_boxed_slice();
+        let boxed_im0 = avec![0.0f64; container_len].into_boxed_slice();
+        let boxed_im1 = avec![0.0f64; container_len].into_boxed_slice();
+
+        Fourier128LweBootstrapKey::from_container(
+            boxed_re0,
+            boxed_re1,
+            boxed_im0,
+            boxed_im1,
+            polynomial_size,
+            input_lwe_dimension,
+            glwe_size,
+            decomposition_base_log,
+            decomposition_level_count,
+        )
+    }
+}
+
+impl<Cont> Fourier128LweBootstrapKey<Cont>
+where
+    Cont: ContainerMut<Element = f64>,
+{
+    /// Fill a bootstrapping key with the Fourier transform of a bootstrapping key in the standard
+    /// domain.
+    pub fn fill_with_forward_fourier<Scalar, ContBsk>(
+        &mut self,
+        coef_bsk: &LweBootstrapKey<ContBsk>,
+        fft: Fft128View<'_>,
+    ) where
+        Scalar: UnsignedTorus,
+        ContBsk: Container<Element = Scalar>,
+    {
+        fn implementation<Scalar: UnsignedTorus>(
```

Comment:
i don't know if this would be better. asserts don't display the function name, and stack traces show the full path

```rust
fn foo() {
    fn implementation() {
        panic!();
    }
    implementation()
}

fn main() {
    foo()
}
```

`implementation` is show as `project_name::foo::implementation`

Reply:
Ah ok, then it's good like that !